storage/io/src/main/java/org/openscience/cdk/io/SDFWriter.java

```diff
@@ -126,6 +132,18 @@ public SDFWriter(Set<String> propertiesToWrite) {
         this(new StringWriter(), propertiesToWrite);
     }
 
+    /**
+     * Ensures a writer is buffered.
+     *
+     * @param wtr writer, may be buffered
+     * @return a BufferedWriter
+     */
+    private static BufferedWriter ensureBuffered(Writer wtr) {
```

Comment:
But this is a writer? Can't write to a reader. But sure, the inputs could take this.